### PR TITLE
Added feature for gravity based embedding - Issue #542

### PR DESCRIPTION
**Pull request by AzureByte**
- Added unit test cases for various embed gravity options
- Added my name to list of contributors

For issue #542

### CODE REVIEW

File: lib/resize.js [hunk 1]
```diff
@@ -186,11 +186,32 @@ function crop (crop) {
...
+  if (!is.defined(embed)) {
+    // Default
+    this.options.embed = gravity.center;
+  } else if (is.integer(embed) && is.inRange(embed, 0, 8)) {
+    // Gravity (numeric)
+    this.options.embed = embed;
+  } else if (is.string(embed) && is.integer(gravity[embed])) {
+    // Gravity (string)
+    this.options.embed = gravity[embed];
+  } else if (is.integer(embed) && embed >= strategy.entropy) {
```
Comment (lovell) [inline lib/resize.js]: The two conditionals related to strategy-based embed gravity can be removed.

File: lib/resize.js [hunk 2]
```diff
@@ -186,11 +186,32 @@ function crop (crop) {
...
+    // Strategy
+    this.options.embed = embed;
+  } else if (is.string(embed) && is.integer(strategy[embed])) {
+    // Strategy (string)
+    this.options.embed = strategy[embed];
+  } else {
+    throw is.invalidParameterError('embed', 'valid embed id/name/strategy', embed);
```
Comment (lovell) [inline lib/resize.js]: Please can you add a test for this error path.

File: lib/resize.js [hunk 3]
```diff
@@ -186,11 +186,32 @@ function crop (crop) {
  *     // outputBuffer contains WebP image data of a 200 pixels wide and 300 pixels high
  *     // containing a scaled version, embedded on a transparent canvas, of input.gif
  *   });
- *
+ * @param {String} [embed='centre'] - A member of `sharp.gravity` to embed to an edge/corner or `sharp.strategy` to embed dynamically.
```
Comment (lovell) [inline lib/resize.js]: The reference to strategy can be removed here also.

File: test/fixtures/index.js [hunk 4]
```diff
@@ -84,6 +84,7 @@ module.exports = {
   inputPngBooleanNoAlpha: getPath('bandbool.png'),
   inputPngTestJoinChannel: getPath('testJoinChannel.png'),
   inputPngTruncated: getPath('truncated.png'), // gm convert 2569067123_aca715a2ee_o.jpg -resize 320x240 saw.png ; head -c 10000 saw.png > truncated.png
+  inputPngEmbed: getPath('embedgravitybird.png'),
```
Comment (lovell) [inline test/fixtures/index.js]: Text on the image mentions you own the copyright for this test image. Under what licence is sharp granted use to it?

Comment (AzureByte) [reply test/fixtures/index.js]: I am releasing it to sharp under a Creative Commons Attribution 4.0 International License. Do I need to specify this somewhere in the project files?

Comment (lovell) [reply test/fixtures/index.js]: Great, thank you, please can you add "CC BY 4.0" as an inline comment so it's in the git history.

**Comment by AzureByte** [comment]
@lovell Created a new PR #1038 with the requested changes

**Pull request closed without merging**